### PR TITLE
DSL: remove lang and orphan trn tag

### PR DESCRIPTION
**Pull request by miurahr**
resolve a bug reported in omegat-users ML
https://sourceforge.net/p/omegat/mailman/message/37388173/

Signed-off-by: Hiroshi Miura <miurahr@linux.com>

### CODE REVIEW

File: src/org/omegat/core/dictionaries/LingvoDSL.java [hunk 1]
```diff
@@ -170,6 +170,8 @@ private static String replaceTag(final String line) {
         TAG_REPLACEMENTS.put(Pattern.compile("\\[\\[(?<content>.+?)]]"), "&#91;${content}&#93;");
         TAG_REPLACEMENTS.put(Pattern.compile(Pattern.quote("\\[")), "&#91;");
         TAG_REPLACEMENTS.put(Pattern.compile(Pattern.quote("\\]")), "&#93;");
+        // lang tag
+        TAG_REPLACEMENTS.put(Pattern.compile("\\[lang id=(.+?)]"), "");
```
Comment (amake) [inline src/org/omegat/core/dictionaries/LingvoDSL.java:174]: Wouldn't there be a closing `[/lang]` tag as well?

Also [these docs](http://lingvo.helpmax.net/en/troubleshooting/dsl-compiler/dsl-tags/) say that the attribute is `name` not `id`. We can just catch anything by doing `\\[lang[^]]*]`.

Comment (amake) [reply src/org/omegat/core/dictionaries/LingvoDSL.java:174]: Also we handle `lang` in `ignoreTags` below. If we're going to handle it here instead, we should remove it from `ignoreTags` for clarity.

File: src/org/omegat/core/dictionaries/LingvoDSL.java [hunk 2]
```diff
@@ -220,5 +222,7 @@ private static String replaceTag(final String line) {
         for (String tag : ignoreTags) {
             TAG_REPLACEMENTS.put(Pattern.compile("\\[(?<tag>" + tag + ")](?<content>.+?)\\[/\\k<tag>]"), "${content}");
         }
+        // remove orphan trn tag
+        TAG_REPLACEMENTS.put(Pattern.compile("\\[trn]"), "");
```
Comment (amake) [inline src/org/omegat/core/dictionaries/LingvoDSL.java:226]: Have we seen orphan `trn` in the wild? Do we know why only `trn` needs this treatment?

Comment (miurahr) [reply src/org/omegat/core/dictionaries/LingvoDSL.java:226]: Susan report it in omegat-users ML. I see it in capture.

File: src/org/omegat/core/dictionaries/LingvoDSL.java (hunk 1 above)
Comment (miurahr) [reply src/org/omegat/core/dictionaries/LingvoDSL.java:174]: Does this handle closing tag as well?
`TAG_REPLACEMENTS.put(Pattern.compile("\\[/?lang[^]]*]"), "");`

Comment (miurahr) [reply src/org/omegat/core/dictionaries/LingvoDSL.java]: Also adding test and passed it.

File: src/org/omegat/core/dictionaries/LingvoDSL.java (hunk 2 above)
Comment (miurahr) [reply src/org/omegat/core/dictionaries/LingvoDSL.java]: Any tag of `ignoreTags` may be able to be orphan.

**Comment by miurahr** [comment]
#169 will supersede this.

**Pull request closed without merging**